lib/rules/no-unused-vars.js

```diff
@@ -126,6 +126,22 @@ module.exports = {
         }
 
         /**
+         * Determines if a variable has a sibling rest property
+         * @param {Variable} variable - EScope variable object.
+         * @returns {boolean} True if the variable is exported, false if not.
+         * @private
+         */
+        function hasRestSpreadSibling(variable) {
+            return variable.defs
+              .filter(def => def.name.type === "Identifier")
+              .some(def => (
+                def.node.id &&
+                def.node.id.type === "ObjectPattern" &&
+                def.node.id.properties.some(property => property.type === "ExperimentalRestProperty")
```

Comment:
It's only necessary to check the last property of the declaration here rather than iterating through all the properties, because rest properties are always the last property of a declaration.